activesupport/lib/active_support/core_ext/object/with_options.rb

```diff
@@ -62,6 +62,16 @@ class Object
   #
   # Hence the inherited default for `if` key is ignored.
   #
+  # NOTE: When call a class method with_options, call the method as you are outside of the class.
```

Comment:
Lets rephrase to mention why this is, "you cannot call class methods implicitly inside of with_options, refer the the methods using the class name instead, example: Phone. phone_number_types"